code/modules/mob/living/carbon/carbon.dm

```diff
@@ -1135,85 +1135,129 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
+	return TRUE
 
+/mob/living/carbon/proc/forceFedAttackLog(obj/item/reagent_containers/to_eat, mob/user)
+	add_attack_logs(user, src, "Fed [to_eat]. Reagents: [to_eat.reagents.log_list(to_eat)]", to_eat.reagents.harmless_helper() ? ATKLOG_ALMOSTALL : null)
 
 /*TO DO - If/when stomach organs are introduced, override this at the human level sending the item to the stomach
 so that different stomachs can handle things in different ways VB*/
-/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/toEat, bitesize_override, can_taste_container = TRUE)
-	var/this_bite = bitesize_override ? bitesize_override : toEat.bitesize
-	if(!toEat.reagents)
+/mob/living/carbon/proc/consume(obj/item/reagent_containers/food/to_eat, bitesize_override)
+	var/this_bite = bitesize_override ? bitesize_override : to_eat.bitesize
+	if(!to_eat.reagents)
 		return
 	if(satiety > -200)
-		satiety -= toEat.junkiness
-	if(toEat.consume_sound)
-		playsound(loc, toEat.consume_sound, rand(10,50), 1)
-	if(toEat.reagents.total_volume)
-		if(can_taste_container)
-			taste(toEat.reagents)
-		var/fraction = min(this_bite/toEat.reagents.total_volume, 1)
+		satiety -= to_eat.junkiness
+	if(to_eat.consume_sound)
+		playsound(loc, to_eat.consume_sound, rand(10, 50), TRUE)
+	if(to_eat.reagents.total_volume)
+		taste(to_eat.reagents)
+		var/fraction = min(this_bite / to_eat.reagents.total_volume, 1)
 		if(fraction)
-			toEat.reagents.reaction(src, toEat.apply_type, fraction)
-			toEat.reagents.trans_to(src, this_bite*toEat.transfer_efficiency)
+			to_eat.reagents.reaction(src, REAGENT_INGEST, fraction)
+			to_eat.reagents.trans_to(src, this_bite)
+
+/mob/living/carbon/proc/consume_patch_or_pill(obj/item/reagent_containers/p_or_p, user) // p_or_p = patch or pill
+	// The reason why this is bundled up is to avoid 2 procs that will be practically identical
+	if(!p_or_p.reagents.total_volume)
+		return TRUE // Doesn't have reagents, would be fine to use up
+
+	var/apply_method = "swallow"
+	var/reagent_application = REAGENT_INGEST
+	var/requires_mouth = TRUE
+	var/instant = FALSE
+	var/efficiency = 1
+
+	if(ispatch(p_or_p))
+		apply_method = "apply"
+		reagent_application = REAGENT_TOUCH
+		requires_mouth = FALSE
+		efficiency = 0.5 // Patches aren't that good at transporting reagents into the bloodstream
+		var/obj/item/reagent_containers/patch/patch = p_or_p
+		if(patch.instant_application)
+			instant = TRUE
+
+	if(user != src && !instant)
+		if(requires_mouth && (!head || ismachineperson(src))) // You will not feed the IPC
+			to_chat(user, "<span class='warning'>You cannot feed [src] [p_or_p]!</span>")
+			return FALSE
+		visible_message("<span class='warning'>[user] attempts to force [src] to [apply_method] [p_or_p].</span>")
+		if(!do_after(user, 3 SECONDS, TRUE, src, TRUE))
+			return FALSE
+		forceFedAttackLog(p_or_p, user)
+		visible_message("<span class='warning'>[user] forces [src] to [apply_method] [p_or_p].</span>")
+	else
+		to_chat(user, "You [apply_method] [p_or_p].")
```

Comment:
Used to have the notify span class here, it was intended to keep it without one?

Reply:
This is for messages to yourself, I didn't think it was too bad to have this one without any spans, but I'm open for changing this if you think it would be better!